src/simulation/simulation_problem_results.jl

```diff
@@ -554,151 +490,203 @@ function _read_optimizer_stats(res::SimulationProblemResults, ::Nothing)
+)
+    timestamps = get_timestamps(res)
+    interval = timestamps.step
+    resolution = get_resolution(res)
+    horizon = get_forecast_horizon(res)
+    initial_time = isnothing(initial_time) ? first(timestamps) : initial_time
+    end_time =
+        isnothing(len) ? last(timestamps) + interval - resolution :
+        initial_time + (len - 1) * resolution
+
+    requested_range = initial_time:resolution:end_time
+    available_range =
+        first(timestamps):resolution:(last(timestamps) + (horizon - 1) * resolution)
+    invalid_timestamps = setdiff(requested_range, available_range)
+
+    if !isempty(invalid_timestamps)
+        msg = "Requested time does not match available results"
+        @error msg
+        throw(IS.InvalidValue(msg))
+    end
+
+    return requested_range
+end
+
+struct RealizedMeta
+    initial_time::Dates.DateTime
+    resolution::Dates.TimePeriod
+    count::Int
+    start_offset::Int
+    end_offset::Int
+    interval_len::Int
+end
+
+function RealizedMeta(
+    res::SimulationProblemResults;
+    initial_time::Union{Nothing, Dates.DateTime} = nothing,
+    len::Union{Int, Nothing} = nothing,
+)
+    existing_timestamps = get_timestamps(res)
+    interval = existing_timestamps.step
+    resolution = get_resolution(res)
+    interval_len = Int(interval / resolution)
+    realized_timestamps =
+        get_realized_timestamps(res, initial_time = initial_time, len = len)
+
+    result_initial_time = existing_timestamps[findlast(
+        x -> x .<= first(realized_timestamps),
+        existing_timestamps,
+    )]
+    result_end_time = existing_timestamps[findlast(
+        x -> x .<= last(realized_timestamps),
+        existing_timestamps,
+    )]
+
+    count = length(result_initial_time:interval:result_end_time)
+
+    start_offset = length(result_initial_time:resolution:first(realized_timestamps))
+    end_offset = length(
+        (last(realized_timestamps) + resolution):resolution:(result_end_time + interval - resolution),
+    )
+
+    return RealizedMeta(
+        result_initial_time,
+        resolution,
+        count,
+        start_offset,
+        end_offset,
+        interval_len,
+    )
+end
+
+function get_realization(
+    result_values::Dict{
+        OptimizationContainerKey,
+        SortedDict{Dates.DateTime, DataFrames.DataFrame},
+    },
+    meta::RealizedMeta,
+    timestamps,
+)
+    realized_values = Dict{OptimizationContainerKey, DataFrames.DataFrame}()
+    for (key, result_value) in result_values
+        results_concat = Dict{Symbol, Vector{Float64}}()
+        for (step, (t, df)) in enumerate(result_value)
+            first_id = step > 1 ? 1 : meta.start_offset
+            last_id =
+                step == meta.count ? meta.interval_len - meta.end_offset : meta.interval_len
+            result_length = length(first_id:last_id)
+            for colname in propertynames(df)
+                colname == :DateTime && continue
+                # TODO DT: may not be correct
```

Comment:
Can we confirm this and then delete the comment?

Reply:
I can't confirm it is correct at the moment. @claytonpbarrows was assigned the validation. We should leave the comment and merge to dev for further testing of the results code.